File: cases/taint/java/return-relay-one-hop-positive/ReturnRelayOneHopPositive.java
package dataflowbench.taint;

final class ReturnRelayOneHopPositive {
    static int dfb_source() { // DFB-SOURCE: return-one-hop-input
        return 1;
    }

    static int relay(int value) { // DFB-WITNESS: return-one-hop-relay
        return value;
    }

    static void dfb_sink(int value) { } // DFB-SINK: return-one-hop-sink

    static void run() {
        int result = relay(dfb_source());
        dfb_sink(result);
    }
}

File: cases/taint/scala/return-relay-one-hop-positive/ReturnRelayOneHopPositive.scala
package dataflowbench

object ReturnRelayOneHopPositive {
  def dfb_source(): Int = { // DFB-SOURCE: return-one-hop-input
    1
  }

  def relay(value: Int): Int = { // DFB-WITNESS: return-one-hop-relay
    value
  }

  def dfb_sink(value: Int): Unit = {} // DFB-SINK: return-one-hop-sink

  def run(): Unit = {
    val result = relay(dfb_source())
    dfb_sink(result)
  }
}
